deploy/setup/setup.sh

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env bash
+set -e
+
+# --- Parse CLI Arguments ---
+
+for arg in "$@"; do
+  case "$arg" in
+    --cloud-init)
+      CLOUD_INIT=true
+      ;;
+    --dev)
+      SKIP_CLONE=true
+      SKIP_DEPENDENCY_INSTALLS=true
+      SKIP_LAUNCH=true
+      # Root directory above repo
+      WBS_DIR=../../..
+      ;;
+    --local)
+      LOCALHOST=true
+      ;;
+    --skip-clone)
+      SKIP_CLONE=true
+      ;;
+    --skip-deps)
+      SKIP_DEPENDENCY_INSTALLS=true
+      ;;
+    --skip-launch)
+      SKIP_LAUNCH=true
+      ;;
+    --verbose)
+      VERBOSE=true
+      ;;
+    --web)
+      USE_WEB=true
+      ;;
+  esac
+done
+
+# --- Assign Vars with Defaults ---
+
+CLOUD_INIT="${CLOUD_INIT:-false}"
+LOCALHOST="${LOCALHOST:-false}"
+SKIP_CLONE="${SKIP_CLONE:-false}"
+SKIP_DEPENDENCY_INSTALLS="${SKIP_DEPENDENCY_INSTALLS:-false}"
+SKIP_LAUNCH="${SKIP_LAUNCH:-false}"
+REPO_URL="${REPO_URL:-https://github.com/wmde/wikibase-release-pipeline.git}"
+USE_WEB="${USE_WEB:-false}"
+VERBOSE="${VERBOSE:-false}"
+WBS_DIR="${WBS_DIR:-/opt/wbs}"
+DEPLOY_DIR="${DEPLOY_DIR:-$WBS_DIR/wikibase-release-pipeline/deploy}"
+SETUP_DIR="${SETUP_DIR:-$DEPLOY_DIR/setup}"
+SCRIPTS_DIR="$SETUP_DIR/scripts"
+LOG_PATH="${LOG_PATH:-/tmp/wbs-deploy-setup.log}"
+
+# -- Setup logging --
+
+mkdir -p "$(dirname "$LOG_PATH")"
+touch "$LOG_PATH" 2>/dev/null || true
+
+log() {
+  if $VERBOSE; then
+    echo "$@"
+  fi
+}
+
+log_cmd() {
+  if $VERBOSE; then
+    bash -c "$@"
+  else
+    bash -c "$@" &> /dev/null
+  fi
+  return $?
+}
+
+install_git() {
+  if command -v git &>/dev/null; then
+    log "Git already installed"
+    return
+  fi
+
+  echo "Installing Git..."
+  if command -v apt-get &>/dev/null; then
+    log_cmd "apt-get update && apt-get install -y git"
+  elif command -v dnf &>/dev/null; then
+    log_cmd "dnf install -y git"
+  elif command -v yum &>/dev/null; then
+    log_cmd "yum install -y git"
+  elif command -v apk &>/dev/null; then
+    log_cmd "apk add git"
+  elif command -v pacman &>/dev/null; then
+    log_cmd "pacman -Sy --noconfirm git"
+  else
+    echo "\u26a0\ufe0f Unsupported package manager. Please install Git manually."
+    exit 1
+  fi
+}
+
+install_docker() {
+  echo "Installing Docker..."
+  log_cmd "curl -fsSL https://get.docker.com | sh"
```

Comment:
I would definitely prefer to use docker from the official distributions directories here. Have you considered this? Otherwise in my opinion it is quite intrusive to change settings on the users package manager such as adding repos, adding trusted keys etc.

Reply:
Sounds worth looking into. I am not wedded to Docker's convenance script here. I am thinking doing some version of what I am doing above with install_git is in order. I choose for now to go this way as my working reference was Dokploy's own convenance script which *if I remember right*, they used this method. What I do know and like about the convenance script is that I have never seen it fail, and I have ran it across several Linux distros, and it is rather quick. But yes maybe we move to the package managers... Leaving open to loop back to later.